### PR TITLE
fixing the parsing of unit with string "Temperature"

### PR DESCRIPTION
**Pull request by GitPaean**
make sure the offset will not be discarded.

it was spotted when trying to review https://github.com/OPM/opm-simulators/pull/5490 .

### CODE REVIEW

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
Indeed there is some keywords using composite dimensions with Temperature, 

like the following, 

```
input/eclipse/share/keywords/900_OPM/O/OILJT-19-      "value_type": "DOUBLE",
input/eclipse/share/keywords/900_OPM/O/OILJT:20:      "dimension": "Temperature/Pressure",
input/eclipse/share/keywords/900_OPM/O/OILJT-21-      "default": 0,
--
input/eclipse/share/keywords/900_OPM/G/GASJT-19-      "value_type": "DOUBLE",
input/eclipse/share/keywords/900_OPM/G/GASJT:20:      "dimension": "Temperature/Pressure",
input/eclipse/share/keywords/900_OPM/G/GASJT-21-      "default": 0,
--
input/eclipse/share/keywords/900_OPM/W/WATJT-19-      "value_type": "DOUBLE",
input/eclipse/share/keywords/900_OPM/W/WATJT:20:      "dimension": "Temperature/Pressure",
input/eclipse/share/keywords/900_OPM/W/WATJT-21-      "default": 0,
```

**Comment by GitPaean** [comment]
For this cases, they using as   dT/dP, so the offset can be ignored. 

But the PR is trying to fix the situation that 25C will be converted to 25K. 

I think the solution will be to use Absolute temperature for `Temperature/Pressure`.

**Comment by GitPaean** [comment]
jenkins build this please

File: opm/input/eclipse/Units/UnitSystem.cpp [hunk 1]
```diff
@@ -1560,8 +1566,9 @@ namespace {
         Dimension dividend = this->parseFactor( parts[0] );
         Dimension divisor = this->parseFactor( parts[1] );
 
-        if (dividend.getSIOffset() != 0.0 || divisor.getSIOffset() != 0.0)
-            throw std::invalid_argument("Composite dimensions cannot currently require a conversion offset");
+        if (!dividend.isCompositable() || !divisor.isCompositable()) {
```
Comment (atgeirr) [inline opm/input/eclipse/Units/UnitSystem.cpp:1569]: I just understood why this is necessary: If any factor of either dividend or divisor is not compositable, that would usually trigger an exception from parseFactor(), but the exception is if either dividend or divisor is a single incompositable factor, then it must be handled here.

Can you confirm this is a correct understanding?

Comment (GitPaean) [reply opm/input/eclipse/Units/UnitSystem.cpp:1569]: `dividend` and `divisor` themselves can be possibly a single atomic unit with conversion offset. In this PR, this unit can be `Temperature`. 

Function paserFactor can parse `Temperature` while not `Temperature*Pressure` (throw).  

And we do not allow either `dividend` or `divisor` contain conversion offset (like `Temperature`. ) before we think  it is possible.

Comment (GitPaean) [reply opm/input/eclipse/Units/UnitSystem.cpp:1569]: > Can you confirm this is a correct understanding?

Your understanding is correct.

Comment (atgeirr) [reply opm/input/eclipse/Units/UnitSystem.cpp:1569]: Thanks, merging.

**Pull request merged**